pythonx/jedi_vim.py

```diff
@@ -158,42 +158,84 @@ def wrapper(*args, **kwargs):
     return func_receiver
 
 
+# XXX: might use lru cache, given that you can configure it per buffer?!
 current_environment = (None, None)
 
 
+def set_environment(executable=None, bufnr=None):
```

Comment:
Needs support for tab, window, and bufnr.
Could be `scope`.

Reply:
While you can do that, I don't think this is what most users need/want. The scope most users want to use is probably global, but I'm not against it - I'm just saying that you're probably doing more work than necessary if you use a bufnr scope. :)

Reply:
Yeah, it is not really necessary, but a good example for getting the scoping right from the beginning.
More important would be a tab-local setting then I guess.
This is kind of useful in an IDE-like environment where you would work on different projects/venvs in parallel - but yes, it is a bit constructed.